HafmanWorker.cs

```diff
@@ -0,0 +1,136 @@
+using Hafman;
+
+namespace Havman;
+
+public class HafmanWorker
+{
+	private Dictionary<string, string> codeDict { get; }
+	private Dictionary<string, string> deCodeDict { get; }
+	private List<string> lines { get; }
+
+	public HafmanWorker(Node root, List<string> line)
+	{
+		lines = line;
+		codeDict = new Dictionary<string, string>();
+		deCodeDict = new Dictionary<string, string>();
+		var path = "";
+		while (path != " ")
+		{
+			var newLeaf = findLeaf(root);
+			if (newLeaf.Item2.Length <= 1)
+			{
+				codeDict.Add(newLeaf.Item2, newLeaf.Item1);
+				deCodeDict.Add(newLeaf.Item1, newLeaf.Item2);
+			}
+
+			if (newLeaf.Item1 == "")
+			{
+				path = " ";
+			}
+		}
+	}
+
+	public Tuple<string, string> findLeaf(Node root)
+	{
+		string path = "";
+		var current = root;
+		while (true)
+		{
+			if (current.LeftNode == null)
+			{
+				if (current.RightNode == null)
+				{
+					break;
+				}
+			}
+
+			if (current.LeftNode != null)
+			{
+				current = current.LeftNode;
+				path += 0;
+			}
+			else if (current.RightNode != null)
+			{
+				current = current.RightNode;
+				path += 1;
+			}
+		}
+
+		var ans = new Tuple<string, string>(path, current.Data);
+		if (path == "") return ans;
+		current.setNull();
+		return ans;
+	}
+
+	public void WorkerIncode()
```

Comment:
Passing plain text as a parameter to method: `public void EncodeText(string/List<string> text)` and returning encoded text as a return value. Then, writing it into file with a separate method/class